subcommands/devices/apps_states.go

```diff
@@ -0,0 +1,62 @@
+package devices
+
+import (
+	"fmt"
+	"github.com/fatih/color"
+	"github.com/foundriesio/fioctl/subcommands"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	asListLimit int
+)
+
+func init() {
+	appsStatesCmd := &cobra.Command{
+		Use:   "apps-states <name>",
+		Short: "List states of Apps reported by a device",
+		Run:   doListStates,
+		Args:  cobra.ExactArgs(1),
+	}
+	cmd.AddCommand(appsStatesCmd)
+	appsStatesCmd.Flags().IntVarP(&asListLimit, "limit", "n", 1, "Limit the number of Apps states to display.")
+}
+
+func doListStates(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	states, err := api.DeviceGetAppsStates(factory, args[0])
+	subcommands.DieNotNil(err)
+
+	stateColor := map[string]color.Attribute{
+		"healthy":   color.FgGreen,
+		"unhealthy": color.FgRed,
+	}
+
+	for indx, s := range states.States {
+		if indx >= asListLimit {
+			break
+		}
+		fmt.Printf("Time:\t%s\n", s.DeviceTime)
+		fmt.Printf("Hash:\t%s\n", s.Ostree)
+		for name, state := range s.Apps {
+			color.New(stateColor[state.State]).Printf("\t%s:%s\n", name, state.State)
+			if len(state.Uri) > 0 {
+				fmt.Printf("\t\tURI:\t%s\n", state.Uri)
+			}
+			fmt.Println("\t\tServices:")
+			for _, srv := range state.Services {
+				fmt.Printf("\t\t\t%s:\n", srv.Name)
+				fmt.Printf("\t\t\t\tURI:\t%s\n", srv.ImageUri)
+				fmt.Printf("\t\t\t\tHash:\t%s\n", srv.Hash)
+				color.New(stateColor[srv.Health]).Printf("\t\t\t\tHealth:\t%s\n", srv.Health)
```

Comment:
is this safe in the event we ever have an unknown state? or is the chance of an unknown state not possible?

Reply:
aklite sets this value in any case, however in the case of non-aklite client it may be a random value, so I'll make it more reliable.